src/librustdoc/doctest.rs

```diff
@@ -542,16 +544,40 @@ crate fn make_test(
         prog.push_str(everything_else);
     } else {
         let returns_result = everything_else.trim_end().ends_with("(())");
+        // Give each doctest main function a unique name.
+        // This is for example needed for the tooling around `-Z instrument-coverage`.
+        let inner_fn_name = if let Some(test_id) = test_id {
+            format!("_doctest_main_{}", test_id)
```

Comment:
It looks like the test errors (snake case) are due to a double-underscore after `_doctest_main`, which probably means `test_id` can start with an underscore.

Maybe you are converting a path with a leading slash and the slash becomes a leading underscore, or something like that. If it's that simple, you can drop the leading underscore either here or where the `test_id` is generated.

Reply:
Ah yes, i rather tried to add a `#[allow()]` directive in there, I didn’t want to go too crazy sanitizing this.